pkg/envoy/lds/listener.go

```diff
@@ -60,50 +46,14 @@ func newOutboundListener(cfg configurator.Configurator) (*xds_listener.Listener,
 		},
 	}
 
-	if cfg.IsEgressEnabled() {
-		err := updateOutboundListenerForEgress(outboundListener, cfg)
-		if err != nil {
-			log.Error().Err(err).Msgf("Error building egress config for outbound listener")
-			// An error in egress config should not disrupt in-mesh traffic, so only log an error
-		}
-	}
-
-	return outboundListener, nil
-}
-
-func updateOutboundListenerForEgress(outboundListener *xds_listener.Listener, cfg configurator.Configurator) error {
-	// When egress, the in-mesh CIDR is used to distinguish in-mesh traffic
-	meshCIDRRanges := cfg.GetMeshCIDRRanges()
-	if len(meshCIDRRanges) == 0 {
-		log.Error().Err(errInvalidCIDRRange).Msg("Mesh CIDR ranges unspecified, required when egress is enabled")
-		return errInvalidCIDRRange
-	}
-
-	var prefixRanges []*xds_core.CidrRange
-
-	for _, cidr := range meshCIDRRanges {
-		cidrRange, err := getCIDRRange(cidr)
-		if err != nil {
-			log.Error().Err(err).Msgf("Error parsing CIDR: %s", cidr)
-			return err
-		}
-		prefixRanges = append(prefixRanges, cidrRange)
-	}
-	outboundListener.FilterChains[0].FilterChainMatch = &xds_listener.FilterChainMatch{
-		PrefixRanges: prefixRanges,
-	}
-
-	// With egress, a filter chain to match TLS traffic is added to the outbound listener.
-	// In-mesh traffic will always be HTTP so this filter chain will not match for in-mesh.
-	// HTTPS egress traffic will match this filter chain and will be proxied to its original
-	// destination.
-	egressFilterChain, err := buildEgressFilterChain()
+	serviceFilterChains, err := getFilterChains(catalog, cfg, localSvc)
 	if err != nil {
-		return err
+		log.Error().Err(err).Msgf("Error getting filter chain for Egress")
```

Comment:
Log message needs to be updated to reflect call to `getFilterChains`, which doesn't seem specific to egress.